torch/ao/quantization/fx/prepare.py

```diff
@@ -162,6 +162,16 @@ def is_output_dtype_supported_by_backend(
     return output_dtype is None or \
         output_dtype == node_name_to_target_dtype[node.name]["output_activation_dtype"]
 
+def need_remove_observer(node, modules, node_name_to_target_dtype):
+    """ remove output observer when the node output is quint8
```

Comment:
I think we can give more context on why here

Reply:
we first need to explain when do we remove observers, we remove it when `maybe_make_input_output_share_observers` returns False, which means the function fails to share output observer and input observer, and we can assume that input is not observed and remove the output observer.

When node output is not fp32 and it is placeholder, the input is assumed to be quantized, so it is observed in a different place, rather than not observed, therefore, even `maybe_make_input_output_share_observers` returns False, we still can't remove the observer.

This is a bit complicated, we can discuss offline if needed.

Reply:
i think first we need to check the observer need to be removed or not ( based on the condition ( output node + placeholder) , then check  maybe_make_input_output_share_observers true / false.